src/ReactiveDomain/ReactiveDomain/Logging/EventStoreMessageLogger.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using EventStore.ClientAPI;
+using ReactiveDomain.Bus;
+using ReactiveDomain.EventStore;
+using ReactiveDomain.Messaging;
+
+namespace ReactiveDomain.Logging
+{
+    public class EventStoreMessageLogger :
+        QueuedSubscriber,
+        IHandle<Message>
+    {
+        private readonly IEventStoreConnection _eventStore;
+        private readonly string _streamPrefix;
+        private readonly IGeneralBus _bus;
+
+        public EventStoreMessageLogger(
+            IGeneralBus bus,
+            IEventStoreConnection es,
```

Comment:
you can separate out the target of the log from the wire tap & queue by passing in a func<msg> to wrap the es connection details.
then create something like a LogTarget class that implements IHandle<Msg> use that in bootstrapper 
this would be a good target for the refactor of this.
it allows multiple log targets to be swapped, and easy testing

Reply:
I like that idea.